core/templates/services/html-length.service.ts

```diff
@@ -50,51 +50,91 @@ export class HtmlLengthService {
     }
     const sanitizedHtml = this.sanitizer.sanitize(
       SecurityContext.HTML, htmlString) as string;
-    let totalWeight = this.calculateBaselineLength(sanitizedHtml);
+    let totalWords = this.calculateBaselineLength(sanitizedHtml, 'word');
+    // Identify custom tags using regex on the original HTML string.
+    const customTagRegex = /<oppia-noninteractive-(?:math|image)[^>]*>/g;
+    const customTags = htmlString.match(customTagRegex);
+
+    if (customTags) {
+      for (const customTag of customTags) {
+        totalWords += this.getWeightForNonTextNodes(customTag, 'word');
+      }
+    }
+    return totalWords;
+  }
 
+  computeHtmlLengthInCharacters(htmlString: string): number {
+    if (!htmlString) {
+      this.loggerService.error('Empty string was passed to compute length');
+      return 0;
+    }
+    const sanitizedHtml = this.sanitizer.sanitize(
+      SecurityContext.HTML, htmlString) as string;
+    let totalCharacters =
+    this.calculateBaselineLength(sanitizedHtml, 'character');
     // Identify custom tags using regex on the original HTML string.
     const customTagRegex = /<oppia-noninteractive-(?:math|image)[^>]*>/g;
     const customTags = htmlString.match(customTagRegex);
 
     if (customTags) {
       for (const customTag of customTags) {
-        totalWeight += this.getWeightForNonTextNodes(customTag);
+        totalCharacters += this.getWeightForNonTextNodes(
+          customTag, 'character');
       }
     }
-    return totalWeight;
+    return totalCharacters;
   }
 
-  private calculateBaselineLength(sanitizedHtml: string): number {
+  private calculateBaselineLength(
+      sanitizedHtml: string, wordOrCharacter: string): number {
     let domparser = new DOMParser();
     let dom = domparser.parseFromString(sanitizedHtml, 'text/html');
     let totalWeight = 0;
     for (let tag of Array.from(dom.body.children)) {
       const ltag = tag.tagName.toLowerCase();
       if (this.textTags.includes(ltag)) {
-        totalWeight += this.getWeightForTextNodes(tag as HTMLElement);
+        totalWeight +=
+        this.getWeightForTextNodes(tag as HTMLElement, wordOrCharacter);
       }
     }
     return totalWeight;
   }
 
-  private getWeightForTextNodes(textNode: HTMLElement): number {
+  private getWeightForTextNodes(
+      textNode: HTMLElement, wordOrCharacter: string): number {
     const textContent = textNode.textContent || '';
-    const words = textContent.trim().split(' ');
-    const wordCount = words.length;
 
-    return wordCount;
+    if (wordOrCharacter === 'word') {
+      const words = textContent.trim().split(' ');
+      const wordCount = words.length;
+      return wordCount;
+    }
+    const characters = textContent.trim().split('\n').join('').split('');
```

Comment:
Make this an if/else. Or possibly if/elif/throwError. (It depends on how you type the second argument and whether other values besides the given ones are possible inputs.)

Reply:
Done!